### DISCUSSION TITLE
Communication between thread with CSP

### DISCUSSION THREAD

**Discussion by centratelemedia** [activej/activej#200] [Q&A]
in go language we can use go routing and channel to communicate each other,  how we can do the same in activej? for real example

**Comment by eduard-vasinskyi** [comment]
Hi, @centratelemedia
You can communicate between channel suppliers/consumers of another thread (another Eventloop) by creating an adapter for that supplier or consumer. You should use either `ChannelSupplier#ofAnotherEventloop` or `ChannelConsumer#ofAnotherEventloop` method. You can find an example of such between-thread communication by looking at [tests for these methods](https://github.com/activej/activej/blob/6d151e5bbfb7340e5baef07495e9741e3cf4ab75/core-csp/src/test/java/io/activej/csp/ChannelConsumerTest.java#L152-L184).

**Reply by centratelemedia** [reply]
@eduard-vasinskyi , glade hear you again, hopefully will be fine., thank i will check
